docs/internals-mappings.md

```diff
@@ -0,0 +1,428 @@
+        "sdkFieldPath": "Planets"
+      }
+    },
+    {
+      "type": "DirectAssignment",
+      "directAssignment": {
+        "schemaModelName": "SolarSystemResourcePlanetSchema",
+        "schemaFieldPath": "Name",
+        "sdkModelName": "Planet",
+        "sdkFieldPath": "Name"
+      }
+    },
+    {
+      "type": "ModelToModel",
+      "modelToModel": {
+        "schemaModelName": "SolarSystemResourceSchema",
+        "sdkModelName": "SolarSystem",
+        "sdkFieldName": "Properties"
+      }
+    }
+  ],
+  "modelToModelMappings": [
+    {
+      "schemaModelName": "SolarSystemResourceSchema",
+      "sdkModelName": "SolarSystem"
+    },
+    {
+      "schemaModelName": "SolarSystemResourceSchema",
+      "sdkModelName": "SolarSystemProperties"
+    },
+    {
+      "schemaModelName": "SolarSystemResourcePlanetSchema",
+      "sdkModelName": "Planet"
+    }
+  ],
+  "resourceIdMappings": []
+}
+```
+
+To explain these mappings in-turn:
+
+* `modelToModelMappings[0]` and `modelToModelMappings[1]` - The Terraform Schema Model `SolarSystemResourceSchema` should be mapped onto both the SDK Models `SolarSystem` and `SolarSystemProperties` - allowing the `name`, `description` and `planets` fields to be mapped across onto these models as defined below.
+* `modelToModelMappings[2]` - The Terraform Schema Model `SolarSystemResourceSchema` should be mapped onto the SDK Model `Planet`. This allows `name` within `planet` in the Terraform Schema to be mapped to/from the Planets model.
+* `fieldMappings[0]` - The Terraform Schema Field `name` (represented by the field `Name` within the Schema Model `SolarSystemResourceSchema`) gets mapped onto the SDK Field `Name` within the SDK Model `SolarSystem`.
+* `fieldMappings[1]` - The Terraform Schema Field `description` (represented by the field `Description` within the Schema Model `SolarSystemResourceSchema`) gets mapped onto the SDK Field `Description` within the SDK Model `SolarSystemProperties`.
+* `fieldMappings[2]` - The Terraform Schema Field `planets` (represented by the field `Planets` within the Schema Model `SolarSystemResourceSchema`) gets mapped onto the SDK Field `Planets` within the SDK Model `SolarSystemProperties`. Since this is a list, each item will be mapped across in-turn.``
+* `fieldMappings[3]` - The Terraform Schema Field `name` within `planets` (represented by the field `Name` within the Schema Model `SolarSystemResourcePlanetSchema`) gets mapped onto the SDK Field `Name` within the SDK Model `Planet`.
+
+> **Note:** In this (simplified) example there are no `resourceIdMappings` - however examples of this can be seen above.
+
+This would allow the Terraform Generator to output mapping functions similar to below:
+
+> Whilst mappings are bidirectional (that is Schema Model -> SDK Model and SDK Model -> Schema Model) - to reduce the amount of output only the Schema Model -> SDK Model mappings are shown below - but the inverse can be implied.
```

Comment:
Leaving things to be implied for complex subjects will lead to misinterpretations, misunderstandings and errors as we've learnt. For completeness the SDK Model -> Schema Model mappings should be included.